vamp-server/convert.cpp

```diff
@@ -556,8 +558,15 @@ readInput(string format, RequestOrResponse::Direction direction, bool &eof)
     if (format == "json") {
         string err;
         auto result = readInputJson(direction, err, eof);
-        if (err != "") throw runtime_error(err);
-        else return result;
+        const bool isRecognisedError = !result.success && result.errorText != "";
+        
+        // if the RequestOrResponse (result) has been populated with success=false and an error message
+        // then the server returned a well formed error, it is safe to return it for conversion 
+        // -- but if err is populated, something else has gone wrong
+        if (isRecognisedError || err == "") 
```

Comment:
That suggestion has the same problem. The question is whether you mean

    if (err != "") {
        throw(...);
    } else {
        return result;
    }

or

    if (isRecognisedError) {
        return result;
    } else if (err != "") {
        throw(...);
    } else {
        return result;
    }

The code does the latter, the comment suggests (to me) the former.